js/src/util.ts

```diff
@@ -151,6 +151,18 @@ export function ensureBN(val: unknown): BN {
   throw new Error(`Not an integer: ${val}`)
 }
 
+export function ensureJsonOrParseString(val: unknown): Object {
+  if (typeof val === "string") {
+    return JSON.parse(val)
+  }
+
+  if (typeof val == "object") {
+    return val as Object
+  }
+
+  throw new Error(`Not a string or a json object: ${val}`)
+}
+
```

Comment:
I would suggest something like:

```
export function ensureParsedJsonObject<T extends object = any>(
  val: unknown
): T {
  if (typeof val === "string") {
    return JSON.parse(val)
  }

  if (typeof val === "object") {
    return val as T
  }

  throw new Error(`Not a JSON or an object: ${val}`)
}
```

1. JSON implies text, so `ensureJsonOr` may mislead the developer into thinking that it might return a string
2. We shouldn't use uppercase `Object`, instead a lowercase `object` but make it possible for the user to define the shape of the object if needed or default to any.
3. Always use `===`.

Reply:
Thanks, done c5e0d171e4a6fa75f11ce18f934709047f095034